src/js/engine.js

```diff
@@ -121,7 +122,8 @@ var plateLayOutWidget = plateLayOutWidget || {};
                 var index = this.stackUpWithColor[color][tileIndex]; 
                 var tile = THIS.allTiles[index];
                 var well = this.derivative[index];
-                THIS.setTileColor(tile, color, this.stackPointer); 
+                this.colorMap.set(index, color);
+                THIS.setTileColor(tile, color);
```

Comment:
stackPointer arg should also be removed from setTileColor in create-canvas-elements.js